lib/data/network/upload_file/upload_file_api.dart

```diff
@@ -0,0 +1,33 @@
+
+import 'dart:io';
+
+import 'package:dio/dio.dart';
+import 'package:fluffychat/data/model/upload_file_json.dart';
+import 'package:fluffychat/data/network/dio_client.dart';
+import 'package:fluffychat/data/network/homeserver_endpoint.dart';
+import 'package:fluffychat/data/network/upload_file/file_info.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/di/global/network_di.dart';
+import 'package:fluffychat/utils/matrix_sdk_extensions/matrix_file_extension.dart';
+
+class UploadFileAPI {
+  final DioClient _client = getIt.get<DioClient>(instanceName: NetworkDI.homeDioClientName);
+
+  UploadFileAPI();
+
+  Future<UploadFileResponse> uploadFile({required FileInfo fileInfo}) async {
+    final dioHeaders = _client.getHeaders();
+    dioHeaders[HttpHeaders.contentLengthHeader] = await File(fileInfo.filePath).length();
+    dioHeaders[HttpHeaders.contentTypeHeader] = fileInfo.mimeType;
+    final response = await _client.post(
+      HomeserverEndpoint.uploadMediaServicePath.generateHomeserverIdentityEndpoint(),
+      data: fileInfo.readStream ?? File(fileInfo.filePath).openRead(),
```

Comment:
IMO: Pls write unit test for this case.
- fileInfo.readStream != null 
- fileInfo.readStream == null => fileInfo.readStream how do it work?

Reply:
so it will open a Stream from file if readStream is null